java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -42,11 +49,148 @@
     NativeDepsLoader.loadNativeDeps();
   }
 
+  protected static class NestedHostColumnVector {
+    private HostMemoryBuffer data;
+    private HostMemoryBuffer offsets;
+    private HostMemoryBuffer validity;
+    private DType type;
+    private long rows;
+    private Optional<Long> nullCount = Optional.empty();
+    private List<NestedHostColumnVector> nestedChildren = new ArrayList<>();
+
+    public NestedHostColumnVector(DType type, long rows,
+                                  Optional<Long> nullCount, HostMemoryBuffer data, HostMemoryBuffer validity,
+                                  HostMemoryBuffer offsets, List<NestedHostColumnVector> nestedChildren) {
+      this.data = data;
+      this.offsets = offsets;
+      this.validity = validity;
+      this.type = type;
+      this.rows = rows;
+      this.nullCount = nullCount;
+      this.nestedChildren = nestedChildren;
+    }
+    /*
+    protected boolean cleanImpl(boolean logErrorIfNotClean) { // make recursive
+     */
+
+    private HostMemoryBuffer getData() {
+      HostMemoryBuffer ret = null;
+      if (nestedChildren.size() > 0)
+        return nestedChildren.get(0).getData();
+      else return data;
+    }
+
+    private List<HostMemoryBuffer> getValids() {
+      List<HostMemoryBuffer> ret = new ArrayList<>();
+      ret.add(validity);
+      for (NestedHostColumnVector child : nestedChildren) {
+        ret.addAll(child.getValids());
+      }
+      return ret;
+    }
+
+    private List<HostMemoryBuffer> getOffsets() {
+      List<HostMemoryBuffer> ret = new ArrayList<>();
+      ret.add(offsets);
+      for (NestedHostColumnVector child : nestedChildren) {
+        ret.addAll(child.getOffsets());
+      }
+      return ret;
+    }
+
+    private List<Long> getRows() {
+      List<Long> ret = new ArrayList<>();
+      ret.add(rows);
+      for (NestedHostColumnVector child : nestedChildren) {
+        ret.addAll(child.getRows());
+      }
+      return ret;
+    }
+
+    private List<DType> getTypes() {
+      List<DType> ret = new ArrayList<>();
+      ret.add(type);
+      for (NestedHostColumnVector child : nestedChildren) {
+        ret.addAll(child.getTypes());
+      }
+      return ret;
+    }
+
+    public Object getElement(int rowIndex) throws IOException {
```

Comment:
This is horribly inefficient. We need a different API for exporting data.

Reply:
@revans2 , do u have any suggestions on how to pull back to host efficiently for a list?

Reply:
The main thing is to put comments on the public APIs that are horribly bad.  these are ones that are going to create objects which means GC issues. This API creates lots of objects.

Doing something efficient would require more knowledge about memory layout so that we can copy large blocks of data at once.  This we can do with Unsafe and copy the data to an array, which is probably the preferred option if we want to do something performant.  But for now, just put in the comments and we can have follow on work to expose fast APIs.

Reply:
Can you point me to the java code that might be doing something analogous to allow columnar to row to work in a performant way? We can keep these getters internal for testing and add something that would be more performant if you share some pointers. Thanks!

Reply:
Columnar to row is going to require lots of investigation and performance tuning.  Please don't try to boil the ocean here.  Lets just put the comment on it and file a follow on issue.

Reply:
Sounds good. I was not sure if it was a Must-have for this PR. Thanks!